warehouse/clickhouse/clickhouse.go

```diff
@@ -142,15 +150,34 @@ func loadConfig() {
 
 }
 
+func registerTLSConfig(key string, certificate string) {
+	tlsConfig := &tls.Config{}
+	caCert := []byte(certificate)
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCert)
+	tlsConfig.RootCAs = caCertPool
+	clickhouse.RegisterTLSConfig(key, tlsConfig)
+}
+
 // getConnectionCredentials gives clickhouse credentials
 func (ch *HandleT) getConnectionCredentials() credentialsT {
-	return credentialsT{
-		host:     warehouseutils.GetConfigValue(host, ch.Warehouse),
-		dbName:   warehouseutils.GetConfigValue(dbName, ch.Warehouse),
-		user:     warehouseutils.GetConfigValue(user, ch.Warehouse),
-		password: warehouseutils.GetConfigValue(password, ch.Warehouse),
-		port:     warehouseutils.GetConfigValue(port, ch.Warehouse),
-	}
+	tlsName := ""
+	certificate := warehouseutils.GetConfigValue(caCertificate, ch.Warehouse)
+	if len(strings.TrimSpace(certificate)) != 0 {
+		tlsName = "tls"
```

Comment:
`tls` name is super generic. These will be registered in static context. Let's construct tls name with destination id or something, so that it will not clash between multiple destinations of multiple customers like in case of hosted.

Reply:
nice catch